### PR TITLE
audit: whitelist pipe, redirect etc in the system check

### PR DESCRIPTION
**Pull request by xu-cheng**
Per https://github.com/Homebrew/homebrew/issues/37515#issuecomment-77784179

cc @mikemcquaid @jacknagel

### CODE REVIEW

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -714,8 +714,10 @@ def audit_line(line, lineno)
     if @strict
       if line =~ /system (["'][^"' ]*(?:\s[^"' ]*)+["'])/
         bad_system = $1
-        good_system = bad_system.gsub(" ", "\", \"")
-        problem "Use `system #{good_system}` instead of `system #{bad_system}` "
+        unless %w[| < > & ;].any? { |c| bad_system.include? c }
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: Could use a regex here rather than `.any?`

**Comment by mikemcquaid** [comment]
Yeh, this is great, thanks @xu-cheng (and sorry @jacknagel)

File: Library/Homebrew/cmd/audit.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: I did use regex at first. But I found it has poor readability.

Comment (mikemcquaid) [inline Library/Homebrew/cmd/audit.rb]: I guess at this stage it's running on only `system` lines anyway so seems fine to me

**Pull request closed without merging**